e2e flake: the object has been modified; please apply your changes to the latest version and try again

Seen randomly when running locally:

```
=== NAME  TestSuites/Exhaustive/Deploy-cephfs#01/Appset/Enable
    actions_test.go:31: Operation cannot be fulfilled on placements.cluster.open-cluster-management.io "appset-deploy-cephfs-busybox": the object has been modified; please apply your changes to the latest version and try again
=== NAME  TestSuites/Exhaustive/Deploy-cephfs#01/Appset
    exhaustive_suite_test.go:93: Enable failed
...
        --- FAIL: TestSuites/Exhaustive/Deploy-cephfs#01 (0.00s)
            --- FAIL: TestSuites/Exhaustive/Deploy-cephfs#01/Appset (0.16s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs#01/Appset/Deploy (0.13s)
                --- FAIL: TestSuites/Exhaustive/Deploy-cephfs#01/Appset/Enable (0.03s)

```

Looks like we need to use `retry.RetryOnConflict` as we use in the unit tests.

--- a/Makefile
+++ b/Makefile
@@ -114,15 +114,20 @@ generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
 
-.PHONY: lint
-lint: golangci-bin ## Run configured golangci-lint and pre-commit.sh linters against the code.
 # golangci-lint has a limitation that it doesn't lint subdirectories if
 # they are a different module.
 # see https://github.com/golangci/golangci-lint/issues/828
+
+.PHONY: lint
+lint: golangci-bin lint-e2e lint-api ## Run configured golangci-lint and pre-commit.sh linters against the code.
 	testbin/golangci-lint run ./... --config=./.golangci.yaml
-	cd api && ../testbin/golangci-lint run ./... --config=../.golangci.yaml
-	cd e2e && ../testbin/golangci-lint run ./... --config=../.golangci.yaml
 	hack/pre-commit.sh
+
+lint-e2e: golangci-bin ## Run configured golangci-lint for e2e module
+	cd e2e && ../testbin/golangci-lint run ./... --config=../.golangci.yaml
+
+lint-api: golangci-bin ## Run configured golangci-lint for api module
+	cd api && ../testbin/golangci-lint run ./... --config=../.golangci.yaml
 
 .PHONY: create-rdr-env
 create-rdr-env: drenv-prereqs ## Create a new rdr environment.

--- a/e2e/dractions/actions.go
+++ b/e2e/dractions/actions.go
@@ -43,7 +43,7 @@ func EnableProtection(w workloads.Workload, d deployers.Deployer) error {
 	placementName := name
 	drpcName := name
 
-	placement, placementDecisionName, err := waitPlacementDecision(util.Ctx.Hub.CtrlClient, namespace, placementName)
+	placementDecisionName, err := waitPlacementDecision(util.Ctx.Hub.CtrlClient, namespace, placementName)
 	if err != nil {
 		return err
 	}

--- a/e2e/dractions/actions.go
+++ b/e2e/dractions/actions.go
@@ -165,11 +165,6 @@ func waitAndUpdateDRPC(client client.Client, namespace, drpcName string, action 
 		return err
 	}
 
-	drpc, err := getDRPC(client, namespace, drpcName)
-	if err != nil {
-		return err
-	}
-
 	drPolicyName := util.DefaultDRPolicyName
 
 	drpolicy, err := util.GetDRPolicy(client, drPolicyName)
@@ -182,16 +177,23 @@ func waitAndUpdateDRPC(client client.Client, namespace, drpcName string, action 
 		return err
 	}
 
-	drpc.Spec.Action = action
-	if action == ramen.ActionFailover {
-		drpc.Spec.FailoverCluster = targetCluster
-	} else {
-		drpc.Spec.PreferredCluster = targetCluster
-	}
-
 	util.Ctx.Log.Info("update drpc " + drpcName + " " + strings.ToLower(string(action)) + " to " + targetCluster)
 
-	return updateDRPC(client, drpc)
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		drpc, err := getDRPC(client, namespace, drpcName)
+		if err != nil {
+			return err
+		}
+
+		drpc.Spec.Action = action
+		if action == ramen.ActionFailover {
+			drpc.Spec.FailoverCluster = targetCluster
+		} else {
+			drpc.Spec.PreferredCluster = targetCluster
+		}
+
+		return updateDRPC(client, drpc)
+	})
 }
 
 func GetNamespace(d deployers.Deployer, w workloads.Workload) string {

--- a/e2e/dractions/actions.go
+++ b/e2e/dractions/actions.go
@@ -11,6 +11,7 @@ import (
 	"github.com/ramendr/ramen/e2e/deployers"
 	"github.com/ramendr/ramen/e2e/util"
 	"github.com/ramendr/ramen/e2e/workloads"
+	"k8s.io/client-go/util/retry"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
@@ -55,18 +56,22 @@ func EnableProtection(w workloads.Workload, d deployers.Deployer) error {
 	clusterName := placementDecision.Status.Decisions[0].ClusterName
 	util.Ctx.Log.Info("got clusterName " + clusterName + " from " + placementDecisionName)
 
-	// move update placement annotation after placement has been handled
-	// otherwise if we first add ocm disable annotation then it might not
-	// yet be handled by ocm and thus PlacementSatisfied=false
-	if placement.Annotations == nil {
-		placement.Annotations = make(map[string]string)
-	}
-
-	placement.Annotations[OcmSchedulingDisable] = "true"
-
 	util.Ctx.Log.Info("update placement " + placementName + " annotation")
 
-	if err = updatePlacement(util.Ctx.Hub.CtrlClient, placement); err != nil {
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		placement, err := getPlacement(util.Ctx.Hub.CtrlClient, namespace, placementName)
+		if err != nil {
+			return err
+		}
+
+		if placement.Annotations == nil {
+			placement.Annotations = make(map[string]string)
+		}
+		placement.Annotations[OcmSchedulingDisable] = "true"
+
+		return updatePlacement(util.Ctx.Hub.CtrlClient, placement)
+	})
+	if err != nil {
 		return err
 	}
 

--- a/e2e/dractions/retry.go
+++ b/e2e/dractions/retry.go
@@ -16,23 +16,23 @@ import (
 )
 
 // nolint:gocognit
-// return placement object, placementDecisionName, error
+// return placementDecisionName, error
 func waitPlacementDecision(client client.Client, namespace string, placementName string,
-) (*v1beta1.Placement, string, error) {
+) (string, error) {
 	startTime := time.Now()
 	placementDecisionName := ""
 
 	for {
 		placement, err := getPlacement(client, namespace, placementName)
 		if err != nil {
-			return nil, "", err
+			return "", err
 		}
 
 		for _, cond := range placement.Status.Conditions {
 			if cond.Type == "PlacementSatisfied" && cond.Status == "True" {
 				placementDecisionName = placement.Status.DecisionGroups[0].Decisions[0]
 				if placementDecisionName != "" {
-					return placement, placementDecisionName, nil
+					return placementDecisionName, nil
 				}
 			}
 		}
@@ -41,11 +41,11 @@ func waitPlacementDecision(client client.Client, namespace string, placementName
 		// so need query placementdecision by label
 		placementDecision, err := getPlacementDecisionFromPlacement(client, placement)
 		if err == nil && placementDecision != nil {
-			return placement, placementDecision.Name, nil
+			return placementDecision.Name, nil
 		}
 
 		if time.Since(startTime) > time.Second*time.Duration(util.Timeout) {
-			return nil, "", fmt.Errorf(
+			return "", fmt.Errorf(
 				"could not get placement decision for " + placementName + " before timeout, fail")
 		}
 
@@ -136,7 +136,7 @@ func waitDRPCPhase(client client.Client, namespace, name string, phase ramen.DRS
 }
 
 func getCurrentCluster(client client.Client, namespace string, placementName string) (string, error) {
-	_, placementDecisionName, err := waitPlacementDecision(client, namespace, placementName)
+	placementDecisionName, err := waitPlacementDecision(client, namespace, placementName)
 	if err != nil {
 		return "", err
 	}

